packages/@okta/vuepress-site/docs/guides/authenticators-okta-verify/main/java/commonsteps.md

```diff
@@ -0,0 +1,105 @@
+Since several steps in the enrollment and challenge flows are nearly identical, it's recommended to componentize the logic to reduce duplicate code and increase reusability. Specifically, the common areas include:
+
+* [Initiating sign-in and returning a list of authenticators to the user](#initiate-sign-in-and-return-a-list-of-authenticators)
+* [Polling Okta](#polling-okta)
+
+A description and step-by-step integration for each of these common steps follow.
+
+### Initiate sign-in flow and return a list of authenticators
+
+All four flows start with the same steps that enable a user to sign in with a username and password, and then choose Okta Verify from a list of authenticators. The following diagram shows this common flow:
+
+<div class="common-image-format">
+
+![Sequence diagram showing the shared sign-in initiation flow for Okta Verify challenge and enrollment](/img/authenticators/java-authenticators-okta-verify-shared-code-initiate-signin.png "All steps in the shared sign-in flow")
+
+</div>
+
+#### 1: Authenticate the user credentials
+
+After a user initiates the sign-in flow by submitting their username and password add the following code:
+
+```java
+// create an IDX Authentication Wrapper (can be an application scoped singleton)
+IDXAuthenticationWrapper idxAuthenticationWrapper = new IDXAuthenticationWrapper();
+// begin transaction
+// The proceedContext needs to be persisted between interactions (for example, stored in a HttpSession for web apps)
+ProceedContext proceedContext = idxAuthenticationWrapper.begin().getProceedContext();  // TODO this needs request context
+// set the user's credentials
+AuthenticationOptions authenticationOptions = new AuthenticationOptions(username, password);
+// start authentication flow
+AuthenticationResponse authenticationResponse =
+    idxAuthenticationWrapper.authenticate(authenticationOptions, proceedContext);
+```
+
+#### 2: Handle the response from the sign-in flow
+
+The `AuthenticationResponse` contains the current state of the authentication process. It could be a `SUCCESS` or may indicate that more information is needed to complete the flow, for example, require an additional factor.
```

Comment:
```suggestion
The `AuthenticationResponse` contains the current state of the authentication process. It could be a `SUCCESS` or indicate that more information is needed to complete the flow, for example, requiring an additional factor.
```